### PR TITLE
Build: Migrate mono-repo dependency versions to use yarn workspace syntax

### PR DESCRIPTION
**Pull request by valentinpalkovic**
Closes N/A

## What I did

- Migrated mono-repo dependency versions to use yarn workspace syntax

## Checklist

- [ ] Make sure your changes are tested (stories and/or unit, integration, or end-to-end tests)
- [ ] Make sure to add/update documentation regarding your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "build", "documentation", "maintenance", "dependencies", "other"]`

This pull request has been released as version [`7.2.0-canary-23572-1690275421-726e1551.0`](https://npmjs.com/package/@storybook/cli/v/7.2.0-canary-23572-1690275421-726e1551.0). Install it by pinning all your Storybook dependencies to that version.
<details>
<summary>More information</summary>

| | |
| --- | --- |
| **Published version** | [`7.2.0-canary-23572-1690275421-726e1551.0`](https://npmjs.com/package/@storybook/cli/v/7.2.0-canary-23572-1690275421-726e1551.0) |
| **Triggered by** | @valentinpalkovic |
| **Repository** | [storybookjs/storybook](https://github.com/storybookjs/storybook) |
| **Branch** | [`valentin/migrate-version-to-use-workspace-syntax`](https://github.com/storybookjs/storybook/tree/valentin/migrate-version-to-use-workspace-syntax) |
| **Commit** | [`726e1551`](https://github.com/storybookjs/storybook/commit/726e1551d87555a7869815c033e22a5db2864bc2) |
| **Datetime** | Tue Jul 25 08:57:01 UTC 2023 (`1690275421`) |
| **Workflow run** | [5654725742](https://github.com/storybookjs/storybook/actions/runs/5654725742) |

To request a new release of this pull request, mention the `@storybookjs/core` team.

_core team members can create a new canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=23572`_
</details>

### CODE REVIEW

**Comment by ndelangen** [comment]
Let's discuss today @valentinpalkovic

File: scripts/release/publish.ts [hunk 1]
```diff
@@ -124,7 +124,7 @@ const publishAllPackages = async ({
   dryRun?: boolean;
 }) => {
   console.log(`📦 Publishing all packages...`);
-  const command = `yarn workspaces foreach --parallel --no-private --verbose npm publish --tolerate-republish --tag ${tag}`;
+  const command = `yarn workspaces foreach --parallel --no-private --verbose exec "yarn pack && npm publish ./package.tgz --tolerate-republish --tag ${tag} && rm package.tgz"`;
```
Comment (JReinhold) [inline scripts/release/publish.ts:127]: I'm pretty sure this change would mean that we need to change the `YARN_NPM_AUTH_TOKEN` env var to just `NPM_TOKEN` [here](https://github.com/storybookjs/storybook/blob/valentin/migrate-version-to-use-workspace-syntax/.github/workflows/publish.yml#L107).

**Review by JReinhold — CHANGES_REQUESTED** [review]
> yarn npm publish would call internally yarn pack. So this would be a win. Unfortuately, it doesn't support the same CLI arguments then npm publish. Therefore first yarn pack is called to create a package.tgz per package, and with npm publish package.tgz this package is taken to publish to npm. Therefore, npm isn't responsible anymore to build the package. It only takes the tarball archive and publishes it.

I don't really follow this. If `yarn npm publish` already runs `yarn pack`, why do you want to run `npm publish` separately? You say that _"it doesn't support the same CLI arguments then npm publish"_, but as far as I can see you don't supply any other arguments other than the `.tgz` file, which I would think wouldn't be needed at all if you just used `yarn npm publish`?
What I'm saying is, isn't `yarn pack && npm publish ./package.tgz --tolerate-republish --tag ${tag}` the exact same thing as `yarn npm publish --tolerate-republish --tag ${tag}`?

**Comment by valentinpalkovic** [comment]
> 

You're right! For the actual release, we might be able to switch to `yarn npm publish`. But I was referencing this code https://github.com/storybookjs/storybook/pull/23572/files#diff-f350f808916daeed26ed59e8d3d6be02aff71db58dd04ce8dcda9ae280dbb537L73-R73. So for verdaccio it seemed to be complicated to switch to `yarn npm publish`. I tried it, but already setting up the Verdaccio registry seemed to not be worth to switch to `yarn npm publish`.

**Comment by JReinhold** [comment]
> You're right! For the actual release, we might be able to switch to `yarn npm publish`. But I was referencing this code [#23572 (files)](https://github.com/storybookjs/storybook/pull/23572/files#diff-f350f808916daeed26ed59e8d3d6be02aff71db58dd04ce8dcda9ae280dbb537L73-R73). So for verdaccio it seemed to be complicated to switch to `yarn npm publish`. I tried it, but already setting up the Verdaccio registry seemed to not be worth to switch to `yarn npm publish`.

Oh I see what you mean. Yeah all of this makes sense to me.

**Review by ndelangen — APPROVED** [review]

**Review by JReinhold — APPROVED** [review]

**Pull request merged**